### PR TITLE
Add com.gitlab.librebob.Athenaeum

### PR DESCRIPTION
**Pull request by librebob**
https://gitlab.com/librebob/athenaeum

### CODE REVIEW

**Comment by librebob** [comment]
bot, build com.gitlab.librebob.Athenaeum

**Comment by librebob** [comment]
bot, build com.gitlab.librebob.Athenaeum

**Comment by nedrichards** [comment]
I get this crash when running after install:
> Traceback (most recent call last):
  File "/app/lib/python3.7/site-packages/peewee.py", line 2875, in execute_sql
    cursor.execute(sql, params or ())
sqlite3.IntegrityError: NOT NULL constraint failed: metarecord.value

> During handling of the above exception, another exception occurred:

> Traceback (most recent call last):
  File "/app/lib/python3.7/site-packages/athenaeum-0.1-py3.7.egg/athenaeum/library.py", line 30, in load
    self.updateFilters(True)
  File "/app/lib/python3.7/site-packages/athenaeum-0.1-py3.7.egg/athenaeum/library.py", line 269, in updateFilters
    self.filterGames(self.filterValue)
  File "/app/lib/python3.7/site-packages/athenaeum-0.1-py3.7.egg/athenaeum/library.py", line 242, in filterGames
    setMeta(key='filter', value=filter)
  File "/app/lib/python3.7/site-packages/athenaeum-0.1-py3.7.egg/athenaeum/models.py", line 39, in setMeta
    MetaRecord.insert(key=key, value=value).on_conflict(action='REPLACE').execute()
  File "/app/lib/python3.7/site-packages/peewee.py", line 1737, in inner
    return method(self, database, *args, **kwargs)
  File "/app/lib/python3.7/site-packages/peewee.py", line 1808, in execute
    return self._execute(database)
  File "/app/lib/python3.7/site-packages/peewee.py", line 2512, in _execute
    return super(Insert, self)._execute(database)
  File "/app/lib/python3.7/site-packages/peewee.py", line 2275, in _execute
    cursor = database.execute(self)
  File "/app/lib/python3.7/site-packages/peewee.py", line 2888, in execute
    return self.execute_sql(sql, params, commit=commit)
  File "/app/lib/python3.7/site-packages/peewee.py", line 2882, in execute_sql
    self.commit()
  File "/app/lib/python3.7/site-packages/peewee.py", line 2666, in __exit__
    reraise(new_type, new_type(*exc_args), traceback)
  File "/app/lib/python3.7/site-packages/peewee.py", line 179, in reraise
    raise value.with_traceback(tb)
  File "/app/lib/python3.7/site-packages/peewee.py", line 2875, in execute_sql
    cursor.execute(sql, params or ())
peewee.IntegrityError: NOT NULL constraint failed: metarecord.value

**Comment by librebob** [comment]
bot, build com.gitlab.librebob.Athenaeum

**Comment by librebob** [comment]
@nedrichards fixed upstream.

File: com.gitlab.librebob.Athenaeum.json [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+                }
+            ]
+        },
+        {
+            "name": "athenaeum",
+            "buildsystem": "simple",
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://gitlab.com/librebob/athenaeum.git"
```
Comment (nedrichards) [inline com.gitlab.librebob.Athenaeum.json]: This needs to be a specific release or commit, see the requirements here: https://github.com/flathub/flathub/wiki/App-Requirements

**Comment by librebob** [comment]
bot, build com.gitlab.librebob.Athenaeum

File: com.gitlab.librebob.Athenaeum.json [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    "rename-icon": "athenaeum",
+    "rename-appdata-file": "athenaeum.appdata.xml",
+    "rename-desktop-file": "athenaeum.desktop",
+    "finish-args": [
+        "--socket=pulseaudio",
+        "--socket=fallback-x11",
+        "--device=dri",
+        "--share=network",
+        "--filesystem=xdg-data/athenaeum:create",
+        "--filesystem=xdg-data/flatpak:ro",
```
[the remark this answers is not in the document]
Comment (librebob) [reply com.gitlab.librebob.Athenaeum.json]: Creating the local sqlite database. Access to the users flatpak appstream files to read and display the games in the app.

File: com.gitlab.librebob.Athenaeum.json [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://gitlab.com/librebob/athenaeum.git",
+                    "tag": "v0.1.0",
+                    "commit": "fd4a2da501eec2c0af840567268533d9b5f835e2"
+                }
+            ],
+            "build-commands": [
+                "for i in 32 48 64 96 128 256 512; do
```
[the remark this answers is not in the document]
Comment (librebob) [reply com.gitlab.librebob.Athenaeum.json]: My understanding is that things such as app icons and desktop files constitute distro packaging and should be done outside the python distutils. Regardless I couldn't get it working in the setup.py, it places the data into the egg which is inaccessible to the OS shell, etc.

File: com.gitlab.librebob.Athenaeum.json [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.12",
+    "sdk": "org.kde.Sdk",
+    "command": "athenaeum",
+    "rename-icon": "athenaeum",
+    "rename-appdata-file": "athenaeum.appdata.xml",
+    "rename-desktop-file": "athenaeum.desktop",
+    "finish-args": [
+        "--socket=pulseaudio",
+        "--socket=fallback-x11",
```
[the remark this answers is not in the document]
Comment (librebob) [reply com.gitlab.librebob.Athenaeum.json]: I'm not sure PyQt5 supports wayland yet.

[the remark this answers is not in the document]
Comment (barthalion) [reply com.gitlab.librebob.Athenaeum.json]: Then it should be just `x11`.

File: com.gitlab.librebob.Athenaeum.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (barthalion) [reply com.gitlab.librebob.Athenaeum.json]: Fair enough, thanks.

File: com.gitlab.librebob.Athenaeum.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (barthalion) [reply com.gitlab.librebob.Athenaeum.json]: You should already be able to store your database in $XDG_DATA_HOME without any additional permissions.

[the remark this answers is not in the document]
Comment (librebob) [reply com.gitlab.librebob.Athenaeum.json]: Removed athenaeum:create permission.

File: com.gitlab.librebob.Athenaeum.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (librebob) [reply com.gitlab.librebob.Athenaeum.json]: Changed to just `x11`.

File: com.gitlab.librebob.Athenaeum.json [hunk 5]
```diff
@@ -0,0 +1,140 @@
+{
+    "id": "com.gitlab.librebob.Athenaeum",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.12",
+    "sdk": "org.kde.Sdk",
+    "command": "athenaeum",
+    "rename-icon": "athenaeum",
+    "rename-appdata-file": "athenaeum.appdata.xml",
+    "rename-desktop-file": "athenaeum.desktop",
```
[the remark this answers is not in the document]
Comment (librebob) [reply com.gitlab.librebob.Athenaeum.json]: Went ahead and did the icons too.

**Comment by librebob** [comment]
bot, build com.gitlab.librebob.Athenaeum

**Comment by barthalion** [comment]
It doesn't start for me:

```
Traceback (most recent call last):
  File "/app/lib/python3.7/site-packages/peewee.py", line 2817, in connect
    self._state.set_connection(self._connect())
  File "/app/lib/python3.7/site-packages/peewee.py", line 3110, in _connect
    **self.connect_params)
sqlite3.OperationalError: unable to open database file

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/app/bin/athenaeum", line 11, in <module>
    load_entry_point('athenaeum==0.1', 'gui_scripts', 'athenaeum')()
  File "/app/lib/python3.7/site-packages/athenaeum-0.1-py3.7.egg/athenaeum/athenaeum.py", line 32, in main
    initDatabase()
  File "/app/lib/python3.7/site-packages/athenaeum-0.1-py3.7.egg/athenaeum/models.py", line 77, in initDatabase
    db.connect()
  File "/app/lib/python3.7/site-packages/peewee.py", line 2820, in connect
    self._initialize_connection(self._state.conn)
  File "/app/lib/python3.7/site-packages/peewee.py", line 2666, in __exit__
    reraise(new_type, new_type(*exc_args), traceback)
  File "/app/lib/python3.7/site-packages/peewee.py", line 179, in reraise
    raise value.with_traceback(tb)
  File "/app/lib/python3.7/site-packages/peewee.py", line 2817, in connect
    self._state.set_connection(self._connect())
  File "/app/lib/python3.7/site-packages/peewee.py", line 3110, in _connect
    **self.connect_params)
peewee.OperationalError: unable to open database file

**Comment by librebob** [comment]
Turns out it needs the xdg create permission.

**Comment by librebob** [comment]
bot, build com.gitlab.librebob.Athenaeum

**Comment by librebob** [comment]
issue seems to be when the package name changed, had to update path names.

**Comment by librebob** [comment]
bot, build com.gitlab.librebob.Athenaeum

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/com.gitlab.librebob.Athenaeum

**Pull request closed without merging**